src/sys/windows_impl.rs

```diff
@@ -40,9 +41,7 @@ pub fn reflink(from: &Path, to: &Path) -> io::Result<()> {
 
     let dest = AutoRemovedFile::create_new(to)?;
 
-    if src_is_sparse {
-        dest.set_sparse()?;
-    }
+    dest.set_sparse()?;
```

Comment:
Let's add a comment explaining why we set it to sparse unconditionally:

```suggestion
    // Set the destination to be sparse while we clone.
    // Important to avoid allocating zero-backed real storage when cloning
    // below which will just be released when cloning file extents.
    dest.set_sparse()?;
```